flyteadmin/auth/authzserver/metadata_provider.go

```diff
@@ -107,3 +116,29 @@ func NewService(config *authConfig.Config) OAuth2MetadataProvider {
 		cfg: config,
 	}
 }
+
+func sendAndRetryHttpRequest(client *http.Client, url string, retryAttempts int, retryDelay time.Duration) (*http.Response, error) {
+	var response *http.Response
+	var err error
+	err = async.RetryOnSpecificErrorCodes(retryAttempts+1, retryDelay, func() (*http.Response, error) {
+		response, err = client.Get(url)
+		return response, err
+	}, isTransientErrorCode)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return response, errors.New(fmt.Sprint("Failed to get oauth metadata"))
+	}
+
+	return response, nil
+}
+
+func isTransientErrorCode(resp *http.Response) bool {
+	if resp.StatusCode >= 500 && resp.StatusCode <= 599 {
```

Comment:
nit: can we use the named constants from https://go.dev/src/net/http/status.go ?